### PR TITLE
DM-45882: Prevent source cleaning for empty injection cats

### PR DESCRIPTION
**Pull request by leeskelvin**

### CODE REVIEW

**Review by arunkannawadi — APPROVED** [review]
LGTM! Only question is if it also make sense to skip `_check_sources` instead of subjecting an empty catalog through it.

**Comment by leeskelvin** [comment]
I think skipping `_check_sources` is an excellent idea too. Let me push that when I'm back at my desk and re-run Jenkins. Assuming no issues, and if you're happy, I'll go ahead and merge later this evening. Thanks!

**Comment by arunkannawadi** [comment]
Sure, and in addition to Jenkins, I presume you've confirmed explicitly that #41 is addressed fully as well.

**Comment by leeskelvin** [comment]
Yes, this PR does indeed resolve [#41](https://github.com/lsst/source_injection/issues/41).

File: python/lsst/source/injection/inject_base.py [hunk 1]
```diff
@@ -587,6 +592,11 @@ def _check_sources(self, injection_catalog, binary_flags):
         """
         self.config = cast(BaseInjectConfig, self.config)
 
+        # Exit early if there are no sources to inject.
+        if len(injection_catalog) == 0:
+            self.log.info("Catalog checking not applied to empty injection catalog.")
+            return injection_catalog
```
Comment (arunkannawadi) [inline python/lsst/source/injection/inject_base.py:598]: I'm not sure if the `if` branch here will ever be true here given the check and return earlier, but shouldn't harm at least.

Comment (arunkannawadi) [reply python/lsst/source/injection/inject_base.py:598]: Ah, I misread the positions these lines were placed at.

**Pull request merged**